src/WebJobs.Script/Config/ScriptSettingsManager.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    public sealed class ScriptSettingsManager
+    {
+        private static readonly ScriptSettingsManager _instance = new ScriptSettingsManager();
+
+        private static ConcurrentDictionary<string, string> _settings;
+
+        public static ScriptSettingsManager Instance
```

Comment:
This deviates a bit from what we've discussed and makes it impossible to inject other implementations for types referencing this directly.

We can chat about this if needed, but the approach we discussed where we'd expose the same instance registered with DI statically would resolve this.

That would allow us to have something similar to what @brettsam suggested (e.g. a settings manager that reads from a JSON file) that would be used by code referencing the static property as well.